File: src/Application Specific/Classes/Entity Collections/Order/OrderProduct/OrderProduct.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace OrderProducts

Public Class OrderProduct

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "OrderProduct Properties"

      
Private _OrderProductID As Integer = 0
Public Readonly Property OrderProductID() As Integer
	Get 
		Return _OrderProductID
	End Get	
End Property	
Public Writeonly Property SetOrderProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderProductID", Value)
	End Set
End Property


Private _ProductSupplierID As Integer = 0
Public Readonly Property ProductSupplierID() As Integer
	Get 
		Return _ProductSupplierID
	End Get	
End Property	
Public Writeonly Property SetProductSupplierID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ProductSupplierID", Value)
	End Set
End Property


Private _Quantity As Integer = 0
Public Readonly Property Quantity() As Integer
	Get 
		Return _Quantity
	End Get	
End Property	
Public Writeonly Property SetQuantity() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Quantity", Value)
	End Set
End Property

            Private _QuantityReceived As Integer = 0
            Public ReadOnly Property QuantityReceived() As Integer
                Get
                    Return _QuantityReceived
                End Get
            End Property
            Public WriteOnly Property SetQuantityReceived() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuantityReceived", Value)
                End Set
            End Property

            Private _BuyPrice As Double = 0
            Public ReadOnly Property BuyPrice() As Double
                Get
                    Return _BuyPrice
                End Get
            End Property
            Public WriteOnly Property SetBuyPrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_BuyPrice", Value)
                End Set
            End Property

            Private _SellPrice As Double = 0
            Public ReadOnly Property SellPrice() As Double
                Get
                    Return _SellPrice
                End Get
            End Property
            Public WriteOnly Property SetSellPrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SellPrice", Value)
                End Set
            End Property


            Private _OrderID As Integer = 0
            Public ReadOnly Property OrderID() As Integer
                Get
                    Return _OrderID
                End Get
            End Property
            Public WriteOnly Property SetOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderID", Value)
                End Set
            End Property

            Private _DispatchSiteID As Integer = 0
            Public ReadOnly Property DispatchSiteID() As Integer
                Get
                    Return _DispatchSiteID
                End Get
            End Property
            Public WriteOnly Property SetDispatchSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_DispatchSiteID", Value)
                End Set
            End Property

            Private _DispatchWarehouseID As Integer = 0
            Public ReadOnly Property DispatchWarehouseID() As Integer
                Get
                    Return _DispatchWarehouseID
                End Get
            End Property
            Public WriteOnly Property SetDispatchWarehouseID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_DispatchWarehouseID", Value)
                End Set
            End Property




#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/OrderProduct/OrderProduct.cs
using System.Collections;

namespace FSM.Entity
{
    namespace OrderProducts
    {
        public class OrderProduct
        {
            private DataTypeValidator _dataTypeValidator;

            public OrderProduct()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _OrderProductID = 0;

            public int OrderProductID
            {
                get
                {
                    return _OrderProductID;
                }
            }

            public object SetOrderProductID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderProductID", value);
                }
            }

            private int _ProductSupplierID = 0;

            public int ProductSupplierID
            {
                get
                {
                    return _ProductSupplierID;
                }
            }

            public object SetProductSupplierID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ProductSupplierID", value);
                }
            }

            private int _Quantity = 0;

            public int Quantity
            {
                get
                {
                    return _Quantity;
                }
            }

            public object SetQuantity
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Quantity", value);
                }
            }

            private int _QuantityReceived = 0;

            public int QuantityReceived
            {
                get
                {
                    return _QuantityReceived;
                }
            }

            public object SetQuantityReceived
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuantityReceived", value);
                }
            }

            private double _BuyPrice = 0;

            public double BuyPrice
            {
                get
                {
                    return _BuyPrice;
                }
            }

            public object SetBuyPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_BuyPrice", value);
                }
            }

            private double _SellPrice = 0;

            public double SellPrice
            {
                get
                {
                    return _SellPrice;
                }
            }

            public object SetSellPrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SellPrice", value);
                }
            }

            private int _OrderID = 0;

            public int OrderID
            {
                get
                {
                    return _OrderID;
                }
            }

            public object SetOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderID", value);
                }
            }

            private int _DispatchSiteID = 0;

            public int DispatchSiteID
            {
                get
                {
                    return _DispatchSiteID;
                }
            }

            public object SetDispatchSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_DispatchSiteID", value);
                }
            }

            private int _DispatchWarehouseID = 0;

            public int DispatchWarehouseID
            {
                get
                {
                    return _DispatchWarehouseID;
                }
            }

            public object SetDispatchWarehouseID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_DispatchWarehouseID", value);
                }
            }




            
        }
    }
}